drivers/spi/spi_ll_stm32.c

```diff
@@ -514,8 +536,8 @@ static void spi_stm32_complete(const struct device *dev, int status)
 		LL_SPI_ClearFlag_TXTF(spi);
 		LL_SPI_ClearFlag_OVR(spi);
 		LL_SPI_ClearFlag_EOT(spi);
-		LL_SPI_SetTransferSize(spi, 0);
 	}
+	LL_SPI_SetTransferSize(spi, 0);
```

Comment:
This change isn't directly linked to half duplex support. Would you  mind making it a dedicated commit, with proper justification ?